rewatch/src/sourcedirs.rs

```diff
@@ -65,7 +66,7 @@ pub fn print(buildstate: &BuildState) {
     let (dirs, pkgs): (Vec<AHashSet<Dir>>, Vec<AHashMap<PackageName, AbsolutePath>>) = buildstate
         .packages
         .par_iter()
-        .filter(|(_name, package)| !package.is_root)
+        .filter(|(_name, package)| package.is_local_dep && package.source_files.is_some())
```

Comment:
In case there is only a single rescript.json, this will also ensure it gets processed.

Reply:
The goal here is to find all packages except for the root package, why is this behavior changing?

Reply:
oh the root is never a local dep, and we don't want to have any other deps? Maybe you should update the comment if people run into this code later?

Reply:
When scaffolding a new (single) rescript project:

<img width="938" height="684" alt="Image" src="https://github.com/user-attachments/assets/10e37ec1-75c1-474b-937e-5636128fb363" />

it is both root and local, which is fine.
You only need to avoid the root package if it has no source files. So reanalyze can run on it.